.github/workflows/convert.yaml

```diff
@@ -0,0 +1,57 @@
+name: Convert and validate definitions
+
+on:
+  push:
+    branches: ["main"]
+jobs:
+  Convert:
+    name: Convert definitions
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python 🐍
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.9
+
+      - name: Install python dependencies
+        run: pip install pre-commit data-product-definition-tooling
+
+      - name: Generate definitions
+        run: converter src DataProducts
+
+      - name: Commit and push
+        run: |
+          git status
+
+          if [[ -z $(git status -s) ]]; then
+            echo "No changes detected"
+          else
+            set -x
+            git config --local user.name "${BOT_NAME}"
+            git config --local user.email "${BOT_EMAIL}"
+
+            git fetch --depth=1
+            git checkout "${BRANCH}"
+
+            git add .
+            # format files if necessary
+            pre-commit run --all-files || echo
```

Comment:
I assume `|| echo` is to not trigger errors? If that's the case, it should be `|| true` which is more obvious